File: backend/src/main/java/com/cosre/backend/entity/Task.java
package com.cosre.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDateTime;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name = "tasks")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne
    @JoinColumn(name = "milestone_id")
    private Milestone milestone; // Task thuộc Milestone nào (optional, for context)

    @ManyToOne
    @JoinColumn(name = "team_id", nullable = false)
    @JsonIgnoreProperties({"tasks", "members", "milestones", "classRoom"})
    private Team team; // Task được giao cho nhóm nào

    @ManyToOne
    @JoinColumn(name = "assigned_to_id")
    @JsonIgnoreProperties({"tasks", "teams"})
    private User assignedTo; // Người được giao cụ thể trong nhóm (optional)

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private TaskStatus status = TaskStatus.TO_DO; // Mặc định là TO_DO

    private LocalDateTime dueDate;
}

File: frontend/src/types/Task.ts
// frontend/src/types/Task.ts

export enum TaskStatus {
    TO_DO = "TO_DO",
    IN_PROGRESS = "IN_PROGRESS",
    REVIEW = "REVIEW",
    DONE = "DONE",
    CANCELED = "CANCELED"
}

export interface Task {
    id: number;
    title: string;
    description?: string;
    status: TaskStatus;
    dueDate?: string; // ISO Date string
    team: {
        id: number;
        teamName: string;
    };
    assignedTo?: {
        id: number;
        fullName: string;
        email: string;
    };
    milestone?: {
        id: number;
        title: string;
    };
}

// DTO gửi lên khi tạo mới
export interface CreateTaskRequest {
    title: string;
    description?: string;
    teamId: number;
    assignedToId?: number;
    milestoneId?: number;
    dueDate?: string;
}

// DTO gửi lên khi đổi trạng thái
export interface UpdateStatusRequest {
    newStatus: TaskStatus;
}